internal/sbi/processor/nsselection_network_slice_information.go

```diff
@@ -10,89 +10,53 @@
 package processor
 
 import (
-	"encoding/json"
 	"fmt"
 	"math/rand"
 	"net/http"
-	"net/url"
-	"strings"
 
 	"github.com/gin-gonic/gin"
 
 	"github.com/free5gc/nssf/internal/logger"
-	"github.com/free5gc/nssf/internal/plugin"
 	"github.com/free5gc/nssf/internal/util"
 	"github.com/free5gc/openapi"
 	"github.com/free5gc/openapi/models"
 )
 
-// Parse NSSelectionGet query parameter
-func parseQueryParameter(query url.Values) (plugin.NsselectionQueryParameter, error) {
-	var (
-		param plugin.NsselectionQueryParameter
-		err   error
-	)
+type NetworkSliceInformationGetQuery struct {
+	// nolint: lll
+	NfType models.NrfNfManagementNfType `form:"nf-type" binding:"required,oneof=NRF UDM AMF SMF AUSF NEF PCF SMSF NSSF UDR LMF GMLC 5G_EIR SEPP UPF N3IWF AF UDSF BSF CHF NWDAF PCSCF CBCF HSS UCMF SOR_AF SPAF MME SCSAS SCEF SCP NSSAAF ICSCF SCSCF DRA IMS_AS AANF 5G_DDNMF NSACF MFAF EASDF DCCF MB_SMF TSCTSF ADRF GBA_BSF CEF MB_UPF NSWOF PKMF MNPF SMS_GMSC SMS_IWMSC MBSF MBSTF PANF IP_SM_GW SMS_ROUTER"`
 
-	if query.Get("nf-type") != "" {
-		param.NfType = new(models.NfType)
-		*param.NfType = models.NfType(query.Get("nf-type"))
-	}
+	NfId string `form:"nf-id" binding:"required,uuid"`
 
-	param.NfId = query.Get("nf-id")
+	// nolint: lll
+	SliceInfoRequestForRegistration *models.SliceInfoForRegistration `form:"sliceInfoRequestForRegistration" binding:"omitempty"`
 
-	if query.Get("slice-info-request-for-registration") != "" {
-		param.SliceInfoRequestForRegistration = new(models.SliceInfoForRegistration)
-		err = json.NewDecoder(strings.NewReader(
-			query.Get("slice-info-request-for-registration"))).Decode(param.SliceInfoRequestForRegistration)
-		if err != nil {
-			return param, err
-		}
-	}
+	// nolint: lll
+	SliceInfoRequestForPduSession *models.SliceInfoForPduSession `form:"sliceInfoRequestForPduSession" binding:"omitempty"`
```

Comment:
```suggestion
	SliceInfoRequestForPduSession *models.SliceInfoForPduSession `form:"slice-info-request-for-pdu-session" binding:"omitempty"`
```

Reply:
And other form attribute should be checked!

Reply:
Solved!